php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+     * information from the database and populating it into the
+     * Candidate object. Used by singleton method.
+     *
+     * @param CandID $candID The 6 digit Loris CandID
+     *
+     * @return void As a side-effect loads candidate into current
+     *              object
+     * @throws DatabaseException | LorisException
+     */
+    public function select(CandID $candID): void;
+
+
+    /**
+     * Writes a representation of the current state of this object into the
+     * Database.
+     *
+     * @return void
+     */
+    public function update(): void;
+
+    /**
+     * Checks if candidate exist given a candidate ID and PSCID
+     * Used to check that PSCID matches CandID when entered into the boxes
+     * on the candidate_list form
+     *
+     * @param CandID $candID 6 digit Loris CandID
+     * @param string $PSCID  project study center ID
+     *
+     * @return bool Whether that candidate exists in the database.
+     */
+    public static function candidateExists(CandID $candID, string $PSCID): bool;
+
+    /**
+     * Get a list of SessionIDs for which this Candidate has a corresponding
+     * TimePoint.
+     *
+     * @return SessionID[]
+     */
+    public function getListOfTimePoints(): array;
+
+    /**
+     * Returns the sessionID for this Candidate's nth visit
+     *
+     * @param integer $visitNo The Nth visit for a Candidate.
+     *
+     * @return ?SessionID SessionID for visit specified by $visitNo
+     */
+    public function getSessionID(int $visitNo): ?SessionID;
+
+    /**
+     * Manages the parameter_candidate tableso that projects have a way to add
+     * whatever data they want which is outside the scope of LORIS to their
+     * candidates.
+     *
+     * @param string|null $var The type of data about the Candidate
+     *                    to be returned
+     *
+     * @return     mixed The info requested
+     * @deprecated
+     */
+    public function getData(?string $var = null);
+
+
+    /**
+     * Returns the ProjectID for the current candidate
+     *
+     * @return integer
+     */
+    public function getProjectID(): int;
+
+    /**
+     * Returns the name of the candidate's project from the
+     * config file
+     *
+     * @return string Project name
+     */
+    public function getProjectTitle(): string;
+
+    /**
+     * Returns the CandID of this candidate
+     *
+     * @return CandID 6 digit CandID
+     */
+    public function getCandID();
+
+    /**
+     * Returns the PSCID of this candidate
+     *
+     * @return string The PSCID identifier for candidate
+     */
+    public function getPSCID(): string;
+
+    /**
+     * Returns the site of this candidate
+     *
+     * @return Site
+     */
+    public function getCandidateSite(): Site;
+
+
+    /**
+     * Returns the ID of this candidate's site
+     *
+     * @return int The numeric ID of the candidate's site
+     */
+    public function getCenterID(): int;
```

Comment:
I would not have that function.
The user should call $candidate->getRegistrationSite()->getCenterID(); (And I don't think there should be a getCenterID in Site anyways)